src/common/utxobased/network/Socket.ts

```diff
@@ -146,8 +146,8 @@ export function makeSocket(uri: string, config: SocketConfig): Socket {
   }
 
   const onSocketClose = (): void => {
-    const err = error ?? new Error('Socket close')
-    log.warn(`onSocketClose due to ${err.message} with server ${uri}`)
+    const err = error ?? new Error('Socket closed without error')
+    log.warn(`onSocketClose with server ${uri}: ${err.message}`)
```

Comment:
It is never valid to access `error.message` unless we first determine that the error is `instanceof Error`. Your `handleError` signature up above is wrong, since it should be `e: unknown` for the parameter type, and then follow the TypeScript squiggles until you've handled all the cases.

It might make sense to rename `let error: Error | undefined` to `let errorMessage: string | undefined`, and then `handleError` would have code like `errorMessage = e instanceof Error ? e.message : String(e)`. Then `onSocketClose` wouldn't need to call `new Error`, since we are dealing with plain strings now.

Edit: Oh, derp. You end up doing something similar over the next few commits. I still think you should do the `string` thing, and maybe smoosh the error-handling commits together into one.

Reply:
I think keeping the internal state as an `unknown` time is useful for debugging what gets thrown by a breakpoint in `onSocketClose`. Making `onSocketClose` the top level handler of the thrown thing will make it easier to figure out what weird thing happens at a lower level. For example, if we see `'[object Object]'` errors, then we know some weird object is thrown that cannot be converted to a string. And we can debug it by going to this one place in `onSocketClose` to figure out what the object is, instead of trying to search for where the throw occurred.

Reply:
That makes sense.